lldb/tools/lldb-dap/OutputRedirector.cpp

```diff
@@ -59,7 +64,10 @@ Error OutputRedirector::RedirectTo(std::function<void(StringRef)> callback) {
     while (!m_stopped) {
       ssize_t bytes_count = ::read(read_fd, &buffer, sizeof(buffer));
       // EOF detected.
-      if (bytes_count == 0)
+      if (bytes_count == 0 ||
```

Comment:
I can't make this an inline suggestion, but this would flow better if you replaced this with the following code after the -1 check:
```
StringRef data(buffer, bytes_count);
if (data.empty() || (m_stopped && data == kCloseSentinel)) break;
```
I'm also noting this will not always remove the sentinel as it can end up attached to some other output. That could be handled by something like
```
if (m_stopped) data = data.consume_back(kCloseSentinel);
if (data.empty()); break;
```
.. but that has the potential to remove an unrelated piece of text. Since neither of them is perfect, I'll leave it up to you to choose.

Reply:
I used the second suggestion and added some tests to try to validate we don't incorrectly consume any null bytes produced by the debuggee.